\documentclass{article}
\usepackage{fancyhdr}
\usepackage[margin=2.0cm]{geometry}%rounded up from 1.87, just to be safe...
\usepackage{parskip}
\usepackage{float}
%\usepackage{times} %make sure that the times new roman is used
\usepackage{mathptmx}

\usepackage{blindtext}
\title{Supplemental Information 8: Comparisons with Tara Oceans Data}
\date{May 2020}
\author{Simpson, Bettauer et al.}


%      ------ Format Stuff ---------
\newlength{\itemdist}
\setlength{\itemdist}{0.05ex}
\newlength{\headdist}
\setlength{\headdist}{0.04ex}

\newcommand{\R}{\mathbb{R}}


\begin{document}
%\SweaveOpts{concordance=TRUE}
%\pagestyle{fancy}

\maketitle

<<message=FALSE, warning=FALSE>>= 
options(warn = -1)
root <- rprojroot::find_root(".git/index"); source(file.path(root, "src/init.R"))
source(file.path(root, "experiments/exp8-tara-oceans/local-8.R"))
figurefile <- file.path(MAINFIGUREFILE, "tara")
vers <- 1.0
#load raw tara oceans data (mapped to genera)
library(tidyverse)
tara <- read_tsv(file = file.path(ZEN_DATA ,"refined/tara-ocean/miTAG.taxonomic.profiles.release.tsv"))
@



<<>>=
# tarat <-   include_tara( tree, tara )[[1]]
# nc <- ncol(tarat)
# 
# for (i in 1:nrow(tarat)) {
#   tarat[i , (nc+1): (nc+139) ] <- tarat[i, 25:163] / tarat[1, 25:163]
# }
# colnames(tarat)[(nc+1):(nc+139)] <- paste0(colnames(tarat[25:163]), "_Global_freq")
# 
# toriginal <- tarat
#save(tarat, file = "/home/data/refined/reef/R/tara_oceans_tree.june.9.Rdata")
@

Now we prepare the Tara Oceans metadata.
<<>>=
# sets variable tarat (tara tree)
load(file = file.path(ZEN_DATA ,"refined/R/tara_oceans_tree.june.9.Rdata")) 

# load the tara oceans metadata 
# tabs 1 and 8 from the spreadsheet availble at 
# {\tt ocean-microbiome.embl.de\/companion.html}
w1 <- read.csv(file.path(ZEN_DATA ,"refined/tara-oceans-metadata/W1.csv"))
w5 <- read.csv(file.path(ZEN_DATA ,"refined/tara-oceans-metadata/W5.csv"))
w5 <- w5[ 1:(nrow(w5)-2), ] # last two lines are comments

w8 <- read.csv(file.path(ZEN_DATA ,"refined/tara-oceans-metadata/W8.csv"))
w8 <- w8[ 1:(nrow(w8)-2), ] # last two lines are comments

# column 5 of w1 matches column 1 of w8
colnames(w1) <- c("TARA ID", "INSDC", "INSDC Run AC", "ENA", "PANGAEA", "PANGAEA Data",
                        "TARA Station", "Date/Time", "Latitude", "Longtitude", "Depth",
                        "Environment", "Size Fraction Low", "Size Fraction Up", "Biome",
                        "Region", "MRGID")
colnames(w5) <- c("PANGAEA", "X16S.miTAGs", "Richness observed", "Chao1 richness est.", 
                  "Shannon diversity index")
colnames(w8) <- c("PANGAEA", "Mean Date", "Mean Lat", "Mean Long", "Mean Depth[m]",
                  "Mean Temp [C]", "Mean Salinity [PSU]", "Mean Oxygen [umol/kg]",
                  "Mean Nitrates [umol/L]", "NO2 [umol/L]", "PO4 [umol/L]", "NO2NO3 [umol/L]",
                  "SI [umol/L]", "AMODIS:PAR8d,Einsteins/m-2/d-1", "Okubo-Weiss",
                  "Lyapunov_exp.", 
                  "grad_SST_adv",	"retention",	"Mean Depth MLD Sigma [m]*",
                  "Mean Depth Max Fluo [m]*",
                  "Mean Depth Max N2 [m]*",	"Mean Depth Max O2 [m]*", "Mean Depth Min O2 [m]*",
                  "Mean Depth Nitracline [m]*",	"miTAG.SILVA.Taxo.Richness",	
                  "miTAG.SILVA.Phylo.Diversity",
                  "miTAG.SILVA.Chao",	"miTAG.SILVA.ace",	"miTAG.SILVA.Shannon",
                  "OG.Shannon",	"OG.Richness",	"OG.Evenness",	"FC - heterotrophs [cells/mL]",
                  "FC - autotrophs [cells/mL]",	"FC - bacteria [cells/mL]",
                  "FC - picoeukaryotes [cells/mL]",	"minimum generation time [h]")

metatara <- data.frame()
tnames <- colnames(tarat)[25:163]
confused <- c()
for (i in 1:nrow(tarat)) {
  idx1 <- which(w1$`TARA ID`==tnames[i])
  if ((length(idx1)==0) | (length(idx1)>1)) { confused <- c(confused, i) }
  idx5 <- which(as.character(w5$PANGAEA) == as.character(w1$"PANGAEA"[idx1]))
  if ((length(idx5)==0) | (length(idx5)>1)) { confused <- c(confused, i) }
  idx8 <- which(as.character(w8$PANGAEA) == as.character(w1$"PANGAEA"[idx1]))
  if ((length(idx8)==0) | (length(idx8)>1)) { confused <- c(confused, i) }

  metatara <- rbind(metatara, cbind(cbind(w1[idx1, ], w5[idx5, -1]), w8[idx8, -1]))
}
@

Now we will prep the resultant {\tt metatara} data frame for inclusion in heatmaps and other visualizations. 
Columns with a finite number discrete levels will be converted to numeric.
<<>>=
# remove some redundancies

# column, type, purpose
# 8, weird, date time
# 9, 10, numeric, lat long
# 11, integers, depth (?)
# 12, nominal, environment  (HERE)
# 13, IGNORE
# 14, size fraction, integer (HERE)
# 15, biome, nominal (HERE)
# 16, region, nominal (HERE)
# 17, MRGID, nominal (HERE)
# 18, numeric, 16S tags
# 19, 20, 21, , richness, chao1, shannon, numeric
# 22, date, same as 
# 23, 24  same as 9 and 10
# 25 redundant with 11
# 26-36 numeric
to_remove <- c(8, 11, 23, 24, 13, 47, 49 )
metatara <- metatara[, -to_remove]
#save(metatara, file = file.path(ZEN_DATA ,"refined/R/metatara_1.0.RData"))
@

<<>>=
library(ComplexHeatmap)
library(circlize)
require(pals)

load(file = file.path(ZEN_DATA ,"refined/R/metatara_1.0.RData"))
tree <- tarat
bac <- induce_tree(2)  # bac tree
rownames(bac) <- paste(bac$name, bac$tax_id, sep = "_")


g_bac <- bac[, c(7, 8, 25:163)]
colnames(g_bac)[1:2] <- c("Bellairs", "Maycocks")
g_bac <- g_bac[which(bac$rank == "genus"), ]

g_bac <- apply(g_bac, MARGIN=2, FUN = function(c) {log( (c+ 1) / (sum(c) + length(c) ) )})

tmp <- apply(g_bac, MARGIN=1, var)
tmp_order <- order(tmp, decreasing=TRUE) 
top <- get_top_taxa( g_bac, lmt=5)
small_bac <- g_bac[unique(as.vector(top)), ]

mt_all <- prepare_metatara(metatara)


col_fun = colorRamp2(c(min(small_bac), median(small_bac), max(small_bac)), c("blue", "white", "red"))
col_fun(seq(0, 1))

col_environment = c("DCM" = "pink",
                    "DCM + oxy min" = "purple",
                    "MES" = "blue",
                    "MES + oxy min" = "red",
                    "MIX" = "yellow",
                    "SRF" = "orange",
                    "Barbados" = "green"
  )
col_biome <- c("Coastal Biome "="red","Polar Biome" = "green", "Trades Biome"="yellow", "Westerlies Biome"="cyan", "Barbados" = "blue")

col_region <- c( "Indian" = glasbey()[1], "Med Sea" = glasbey()[2], "North Atlantic" = glasbey()[3],
                 "North Pacific" = glasbey()[4], "Red Sea" = glasbey()[5], "South Atlantic" = glasbey()[6],
                 "Southern" = glasbey()[7], "South Pacific" = glasbey()[8], "Barbados"  = glasbey()[9])

col_MRGID <- glasbey()[1:21]
names(col_MRGID)<- c("ANTA",    "ARAB",   "BENG",   "CAMR",   "CARB",   
                "CHIL",   "EAFR",   "FKLD",   "GFST",   "GUIA",   
                "ISSG",   "MEDI",   "MONS",   "NAST-E", "NAST-W", 
                "NPST",   "PEOD",   "PNEC",   "REDS",   "SATL",   
                "SPSG")
  
fh = function(x) hclust(dist(x), method="ward.D2")
cc <- fh(t(small_bac))
# library(seriation)
# o1 = seriate(fh(small_bac), method = "GW")
# o2 = seriate(fh(t(small_bac)), method = "GW")

a1 = HeatmapAnnotation(
      Environment = mt_all[,10], 
      Biome = mt_all[,12],
      Region = factor(mt_all[,13]),
      Depth= mt_all[,20],
      Temp = mt_all[,21],
      Salinity = mt_all[,22],
      Oxygen = mt_all[,23],
      Nitrates = mt_all[,24],
      NO2 = mt_all[,25],
      PO4 = mt_all[,26],
      NO2NO3 = mt_all[,27],
      SI = mt_all[,28],
      #      MRGID = mt_all[,14],
#      X16S.miTAGs =  mt_all[,15],
 #     Richness = mt_all[,16],
#      Chao1 = mt_all[,17 ],
 #     Shannon = mt_all[,18],

#      AMODIS= mt_all[,29],
#      Okubo_Weiss = mt_all[,30],
#      Lyapunov= mt_all[,31],
#      grad_SST_adv = mt_all[,32],
#      retention = mt_all[,33],
#      Depth_MLD_Sigma = mt_all[,34],
#      Depth_Max_Fluo= mt_all[,35],
 #     Depth_Max_N2= mt_all[,36],
#      Depth_Max_O2 = mt_all[,37],
#      Depth_Min_O2 = mt_all[,38],
#      Depth_Nitracline= mt_all[,39],
#      miTAG.Richness = mt_all[,40],
#      miTAG.Diversity = mt_all[,41],
#      miTAG.ace = mt_all[,42],
#     OG.Shannon = mt_all[,43],
#      OG.Richness = mt_all[,44],
#      OG.Evenness= mt_all[,45],
#      FC_heterotrophs = mt_all[,46],
#      FC_autotrophs = mt_all[,47],
#      FC_bacteria  = mt_all[,48],
#      FC_picoeuk. = mt_all[,49],
 #     gen_time = mt_all[,50],
      
      annotation_name_side = "left",
  #    row_names_gp = gpar(fontsize = 7),
      col= list(Environment = col_environment, Biome = col_biome, Region = col_region )
      )



h1 <- Heatmap(small_bac, name = "expr", 
  #      cluster_rows = TRUE,
   #     cluster_columns = TRUE,
        border = TRUE,
        col = col_fun,
        column_title = "Samples", 
        row_title = "Genera",
    #   row_dend_reorder = TRUE,
    #    column_dend_reorder = TRUE,
        
        cluster_rows = fh(small_bac), 
        cluster_columns =cc,
    
        row_names_gp = gpar(fontsize = 7),
        column_names_gp = gpar(fontsize = 5),
        top_annotation = a1,
        raster_quality = 2

        ) 

draw(h1, merge_legend=TRUE)
# 
# decorate_heatmap_body("expr", {
#     i = which(cc$order == 1)
#     x = i/ncol(small_bac)
#     grid.lines(c(x, x), c(0, 1), gp = gpar(lwd = 2, lty = 2))
#     grid.text("Bellairs", x, unit(1, "npc") + unit(-8, "mm"))
# })
# 
# decorate_heatmap_body("expr", {
#     i = which(cc$order == 2)
#     x = i/ncol(small_bac)
#     grid.lines(c(x, x), c(0, 1), gp = gpar(lwd = 2, lty = 2))
#     grid.text("Maycocks", x, unit(1, "npc") + unit(-11, "mm"))
# })

@




\section{Compositional Data}

In this section we explore some of the advice of Quinn et al.\ (2019) {\em GigaScience} for  compositional data (CoDA) methods. 

<<>>=
# install.packages(c("zCompositions", "propr"))
BiocManager::install("ALDEx2")
library(zCompositions); library(ALDEx2); library(propr)

load(file = file.path(ZEN_DATA ,"refined/R/metatara_1.0.RData"))
tree <- tarat  # tarat extends tree to the tara oceans samples
bac <- induce_tree(2)  # bac tree
rownames(bac) <- paste(bac$name, bac$tax_id, sep = "_")
g_bac <- bac[, c(7, 8, 25:163)]
g_bac <- g_bac[which(bac$rank == "genus"), ]

m_bac <- t( as.matrix(g_bac) )

zeros <- c()
for (i in 1:nrow(m_bac)) {
  zeros[i] <- length(which(m_bac[i,] == 0)) / ncol(m_bac) * 100
}

top <- 100
favs <- matrix(nrow = nrow(m_bac), ncol = top)
for (i in 1:nrow(m_bac)) {
  favs[i, ] <- order(m_bac[i, ], decreasing  = TRUE)[1:top]
  cat("\n", favs[i,])
}
all_favs <- unique(as.vector(favs))
s_m_bac <- m_bac[, all_favs]

zeros <- c()
for (i in 1:nrow(s_m_bac)) {
  zeros[i] <- length(which(s_m_bac[i,] == 0)) / ncol(s_m_bac) * 100
}

g_bac_p <- cmultRepl( s_m_bac, output = "p-counts")
hist(as.matrix(g_bac_p) - s_m_bac)
@

We now explore UMAP.
<<>>=
BiocManager::install("M3C")
library(M3C)
source(file.path(root, "src/umap_M3C.R"))

load(file = file.path(ZEN_DATA ,"refined/R/metatara_1.0.RData"))
tree <- tarat
bac <- induce_tree(2)  # bac tree
rownames(bac) <- paste(bac$name, bac$tax_id, sep = "_")
g_bac <- bac[, c(7, 8, 25:163)]
g_bac <- apply(g_bac, MARGIN=2, FUN = function(c) {c / c[1]})
g_bac <- g_bac[which(bac$rank == "genus"), ]
tmp <- apply(g_bac, MARGIN=1, var)
tmp_order <- order(tmp, decreasing=TRUE) 
top <- get_top_taxa( g_bac, lmt=5)
small_bac <- g_bac[unique(as.vector(top)), ]
mt_all <- prepare_metatara(metatara)


umap(small, labels=as.factor(mt_all$Biome), controlscale=TRUE,scale=3, dotsize = 2)
umap(small, labels=as.factor(mt_all$Environment), controlscale=TRUE,scale=3)
umap(small, labels=as.factor(mt_all$Region), controlscale=TRUE,scale=3)

mt_all$"Mean Temp [C]"[1:2] <- c(0,0)
umap(small, labels=scale(as.numeric(mt_all$"Mean Temp [C]")), controlscale = TRUE,scale=2, dotsize = 2)

mt_all$"Mean Nitrates [umol/L]"[1:2] <- c(0,0)
umap(small, labels=scale(as.numeric(mt_all$"Mean Nitrates [umol/L]")), controlscale = TRUE,scale=2, dotsize = 2)
@




\end{document}